sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -114,9 +129,93 @@
         }
         this.tenantId = tenantId;
         this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.certificatePath = certificatePath;
+        this.certificatePassword = certificatePassword;
         this.options = options;
     }
 
+    private ConfidentialClientApplication getConfidentialClientApplication() {
+        if (confidentialClientApplication != null) {
+            return confidentialClientApplication;
+        } else if (clientId == null) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
+                "A non-null value for client ID must be provided for user authentication."));
+        } else {
```

Comment:
Not needed, but given this has a lot of nesting and all other conditionals hit terminal states above could the `else` be removed for a regular code block?

Reply:
removed.